lib/bolt/inventory/group2.rb

```diff
@@ -118,13 +117,14 @@ def add_target(target)
         end
 
         aliases = target['alias']
-        aliases = [aliases] if aliases.is_a?(String)
+        aliases = [aliases] if aliases.is_a?(String) || aliases.is_a?(Concurrent::Delay)
```

Comment:
Example
```yaml
version: 2
targets: 
  - uri: j6gppqkp45z18uo.delivery.puppetlabs.net
    alias: 
      - _plugin: prompt
        message: alias name
      - foo
    config:
      transport: ssh
      ssh:
        host-key-check: false
        user: 
          _plugin: prompt
          message: ssh user
```
alias name: prompt-me
ssh user: root
```
cas@cas-ThinkPad-T460p:~/working_dir/bolt$ bolt command run whoami -t prompt-me
alias name:
ssh user:
Started on j6gppqkp45z18uo.delivery.puppetlabs.net...
Finished on j6gppqkp45z18uo.delivery.puppetlabs.net:
  STDOUT:
    root
Successful on 1 node: j6gppqkp45z18uo.delivery.puppetlabs.net
Ran on 1 node in 0.43 seconds
```

Reply:
This seems like a bad idea. I will continue working on preventing this outside of config.